compose/desktop/desktop/samples/src/jvmMain/kotlin/androidx/compose/desktop/examples/mouseclicks/Main.jvm.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.desktop.examples.mouseclicks
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.size
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.WindowState
+import androidx.compose.ui.window.singleWindowApplication
+import androidx.compose.desktop.*
+import androidx.compose.ui.input.pointer.isSecondaryPressed
+import androidx.compose.ui.input.pointer.isPrimaryPressed
+
+fun main() {
+    singleWindowApplication(
+        title = "Desktop Mouse Clicks",
+        state = WindowState(width = 1024.dp, height = 850.dp)
+    ) {
+        Box(
+            modifier = Modifier.size(200.dp).background(Color.Red)
+                .combinedMouseClickable(
```

Comment:
As an alternative we can pass PointerButtons or PointerKeyModifiers as parameters:
```
.combinedMouseClickable(
    PointerButtons(isPrimaryPressed = true),
    onDoubleClick = { },
) {
  
}

.combinedMouseClickable(
    PointerButtons(isPrimaryPressed = true),
    PointerKeyboardModifiers(isCtrlPressed = true),
    onDoubleClick = { },
) {
  
}
```

I don't know yet, will it be better, just a thought.

Reply:
Another approach:
```
.combinedMouseClickable(
    buttons = { isPrimaryPressed }
    onDoubleClick = { },
) {
  
}

.combinedMouseClickable(
    buttons = { isPrimaryPressed },
    keyboardModifiers = { isCtrlPressed },
    onDoubleClick = { },
) {
  
}
```

Reply:
In your opinion, if users can pass filters for buttons and keyboardModifiers, would they still need to access keyboardModifiers and buttons in callbacks? 

example:
```
.combinedMouseClickable(
    buttons = { isPrimaryPressed },
    keyboardModifiers = { isCtrlPressed },
    onDoubleClick = { },
) { buttons, keyboardModifiers ->
      // As a result:
      // keyboardModifiers.isCtrlPressed == true
      // buttons.isPrimaryPressed == true
      // at the same time buttons.isAltPressed can also be true or buttons.isSecondaryPressed can be true
}
```

Reply:
> keyboardModifiers in callbacks

I don't see where it will be needed.

If the user wants to handle Alt+Ctrl+Click And Ctrl+Click, they probably should use two different `combinedMouseClickable`, one with `isCtrlPressed && isAltPressed`, and one with `isCtrlPressed && !isAltPressed`

Reply:
I wonder, can we support consuming gestures, not raw events. For example:
```
Box(Modifier.combinedClickable(keyboardModifiers = { isCtrlPressed }, onDoubleClick = { action1() }, onClick = { action3() ) {
    Box(Modifier.combinedClickable(keyboardModifiers = { isCtrlPressed && isAltPressed }, onClick = { action2() }) {

    }
}
```

Reply:
As I understand, current implementation (in this PR) supports such an example (excluding the `keyboardModifiers ` filter).
What do you mean by consuming gestures? Do you mean to call `PointerInputChange.consume()` method only when we know that there is a click/doubleClick/longClick, so only after a gesture is completed?

Reply:
> current implementation (in this PR) supports such an example

Yes, we can exclude it via `if` in the current implementation. We just talk about which API is more convenient and less error-prone API.

> What do you mean by consuming gestures

1. If we click on the nested Box, we should perform `action2`
2. If we double-click or long-click on the parent box, we shouldn't perform click on nested box (because the parent box "consumes" double click and long click)
3. if we ctrl+alt click on the nested box, ctrl-click should not happen on the parent box (because the nested box consumes this click)

(I am not sure about which Box which gesture should consume, though)